drainer.go

```diff
@@ -10,17 +13,30 @@ type Drainer struct {
 	Buffer *recordBuffer
 }
 
-func newDrainer(client *kinesis.Kinesis, streamName string) (*Drainer, error) {
-	buffer, err := newRecordBuffer(client, streamName)
+func newDrainer(a *KinesisAdapter, streamName string) {
+	buffer, err := newRecordBuffer(a.Client, streamName)
 	if err != nil {
-		return nil, err
+		logErr(err)
 	}
 
 	d := &Drainer{Buffer: buffer}
-	// Every second, we flush the buffer
 	go d.Drain()
 
-	return d, nil
+	err = createStream(a, streamName)
+	if err != nil {
+		if reqErr, ok := err.(awserr.RequestFailure); ok {
+			if reqErr.Code() == "ResourceInUseException" {
+				a.addDrainer(streamName, d)
+			} else {
+				logErr(err)
+			}
+		} else {
+			logErr(err)
+		}
+	} else {
+		log.Printf("kinesis: need to create stream for %s\n", streamName)
+		go waitForActive(a, d)
```

Comment:
If newDrainer is already running in a new goroutine, does this need to run in a separate goroutine?